march_gait_selection/src/march_gait_selection/dynamic_gaits/moveit_subgait.py

```diff
@@ -38,7 +39,8 @@ def capture_point_cb(self, msg, leg_name):
         :param msg: The message from the capture point topic
         :param leg_name: The name of corresponding move group
         """
-        self._capture_point_msg[leg_name] = msg
+        self._latest_msg_time[leg_name] = msg.header.stamp
+        self._poses[leg_name] = msg.pose
```

Comment:
```suggestion
        self._latest_capture_point_msg_time[leg_name] = msg.header.stamp
        self._capture_point_position[leg_name] = msg.pose.position
```
You could even consider extracting the x and y coordinate from the msg.pose.position directly and storing vectors in your dict. i.e.
```suggestion
        self._latest_msg_time[leg_name] = msg.header.stamp
        self._poses[leg_name] = [msg.pose.position.x, msg.pose.position.y]
```
Whichever you prefer :)

Reply:
I'm not sure whether storing it like a vector would be beneficial, The move group can use the pose structure like this; 
``` self._move_group[leg_name].set_pose_target(pose, end_effector_link='left_foot')``` 
We can also decide to change it later if necessary and indeed more useful to directly store it like a vector